software/deployment/deployment.go

```diff
@@ -457,7 +457,7 @@ func RuntimeUpgrade(id, desiredRuntimeVersion string, client houston.ClientInter
 		return err
 	}
 
-	if deployment.RuntimeVersion == "" || deployment.AirflowVersion != "" {
+	if deployment.RuntimeVersion == "" && deployment.AirflowVersion != "" {
```

Comment:
should we add a test?

Reply:
have updated the existing tests to cover both real-world scenarios, Deployment going from AC -> runtime -> runtime and Deployment going from runtime -> runtime